flaml/automl.py

```diff
@@ -657,8 +658,15 @@ def custom_metric(
                 add it as an additional element in the metric_constraints. Essentially 'pred_time_limit'
                 specifies a constraint about the prediction latency constraint in seconds.
             custom_hp: dict, default=None | The custom search space specified by user
-                Each key is the estimator name, each value is a dict of the custom search space for that estimator. Notice the
-                domain of the custom search space can either be a value of a sample.Domain object.
+                It is nested dict with keys being the estimator names, and values being dicts
+                of per estimator search space. In the per estimator search space dict,
+                the keys are the hyperparameter names, and values are dicts of info ("domain",
+                "init_value", and "low_cost_init_value") about the search space associated with
+                the hyperparameter (i.e., per hyperparameter search space dict). When custom_hp
+                is provided, the built-in search space which is also a nested dict of per estimator
+                search space dict, will be updated with custom_hp. The update follows the vanilla
```

Comment:
not sure what "vanilla nested dict update rule" means.

Reply:
Revised.